src/WinRT.Runtime/Projections/Nullable.cs

```diff
@@ -1,66 +1,171 @@
+        }
+
+        // This method is only for all blittable types. Note: this method  could be an overload of
+        // the generic one below, but using a different one to simplify the reflection lookup in the
+        // fallback case below (so we can use GetMethod and find just the single match we need).
+        private static unsafe int Do_Abi_get_Value_0_Blittable(void* thisPtr, void* result)
+        {
+            if (result is null)
+            {
+                // Immediately return E_POINTER if the target is null
+                return unchecked((int)0x80004003);
+            }
+
+            try
+            {
+                T unboxedValue = (T)global::WinRT.ComWrappersSupport.FindObject<object>(new IntPtr(thisPtr));
+
+                Unsafe.WriteUnaligned(result, unboxedValue);
+
+                return 0;
+            }
+            catch (global::System.Exception __exception__)
+            {
+                Unsafe.WriteUnaligned<T>(result, default);
+
+                global::WinRT.ExceptionHelpers.SetErrorInfo(__exception__);
+                return global::WinRT.ExceptionHelpers.GetHRForException(__exception__);
+            }
+        }
+
+        private static unsafe int Do_Abi_get_Value_0_DateTimeOffset(void* thisPtr, DateTimeOffset* result)
+        {
+            if (result is null)
+            {
+                return unchecked((int)0x80004003);
+            }
+
+            try
+            {
+                T unboxedValue = (T)global::WinRT.ComWrappersSupport.FindObject<object>(new IntPtr(thisPtr));
+
+                Unsafe.WriteUnaligned(result, DateTimeOffset.FromManaged(Unsafe.As<T, global::System.DateTimeOffset>(ref unboxedValue)));
+
+                return 0;
+            }
+            catch (global::System.Exception __exception__)
+            {
+                Unsafe.WriteUnaligned<T>(result, default);
+
+                global::WinRT.ExceptionHelpers.SetErrorInfo(__exception__);
+                return global::WinRT.ExceptionHelpers.GetHRForException(__exception__);
+            }
+        }
+
+        private static unsafe int Do_Abi_get_Value_0<TAbi>(void* thisPtr, out TAbi __return_value__)
+        {
+            T ____return_value__ = default;
+
+            __return_value__ = default;
+
+            try
+            {
+                ____return_value__ = (T)global::WinRT.ComWrappersSupport.FindObject<object>(new IntPtr(thisPtr));
+                __return_value__ = (TAbi)Marshaler<T>.FromManaged(____return_value__);
+            }
+            catch (global::System.Exception __exception__)
+            {
+                global::WinRT.ExceptionHelpers.SetErrorInfo(__exception__);
+                return global::WinRT.ExceptionHelpers.GetHRForException(__exception__);
+            }
+            return 0;
+        }
+
+        internal static unsafe Delegate GetValueDelegateForAbi()
+        {
+            if (typeof(T) == typeof(int) ||

```

Comment:
So because of [this](https://github.com/microsoft/CsWinRT/blob/ae695510af515a373bc010cf46aa188736b2b8de/src/WinRT.Runtime/ComWrappersSupport.cs#L594-L764), I feel like most of the types listed here won't come to this code path.  Do you think having some specific nullable optimizations for each type is better or do you think have a generic version that handles all here is better?  The main difference I see is with the former, we can list them as `UnManagedCallersOnly` whereas here we would need to do `GetfunctionPointerForDelegate`.  Or should we keep both as is, and have this as a fallback in case that code path wasn't used?

Reply:
Oh, that's interesting, I didn't notice that. To clarify, all changes in this PR were solely driven by trying to fix all AOT warnings and allow all of that LINQ expression code to be trimmed out, basically I did the minimum amount of changes to satisfy that. I'm still seeing generic instantiations for all of these types, for instance:

![image](https://github.com/microsoft/CsWinRT/assets/10199417/f5bacf14-c85b-457a-afe3-5d6f7c430b44)

So I'm inclined to say keeping both code paths can't hurt, and it might still help with trimming since it would make those generic instantiations get the specialized paths (even though we're not 100% sure they're actually executed at runtime). I would say perhaps let's just keep this for now and then we can potentially look into removing code once we've double checked that it is never in fact used anywhere, and also that it doesn't reintroduce any trimming regressions? 🤔

What do you think?

Reply:
Sounds good, we can keep both for now.

Reply:
Note that in your sceenshot, that is all BoxedIReference**Array**Impl which is different from `BoxedIReferenceImpl`.  I know we have worked to do with arrays possibly.

Reply:
The type with a single type parameter does seem to only have one instantiation 🤔

![image](https://github.com/microsoft/CsWinRT/assets/10199417/9676e1e8-d64c-4121-88ad-9fea82763f46)

Reply:
For the first set, those come from [here](https://github.com/microsoft/CsWinRT/blob/ae695510af515a373bc010cf46aa188736b2b8de/src/WinRT.Runtime/ComWrappersSupport.cs#L774-L828).  I wasn't sure if I wanted to create specializations for each one so I had created a BoxedValueIreferenceImpl that takes both the type and the ABI type as generics so it was AOT compatible and had implemented it that way.  With your changes, there might be opportunity in a future PR to make it again use the one generic one if that is better.